benchmarker/cmd/ann_benchmark.go

```diff
@@ -0,0 +1,751 @@
+package cmd
+
+import (
+	"context"
+	"encoding/binary"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/google/uuid"
+	"github.com/spf13/cobra"
+	"github.com/weaviate/hdf5"
+	"github.com/weaviate/weaviate-go-client/v4/weaviate"
+	"github.com/weaviate/weaviate-go-client/v4/weaviate/fault"
+	"github.com/weaviate/weaviate/entities/models"
+	weaviategrpc "github.com/weaviate/weaviate/grpc"
+	"google.golang.org/grpc"
+)
+
+// Batch of vectors and offset for writing to Weaviate
+type Batch struct {
+	Vectors [][]float32
+	Offset  int
+}
+
+// Weaviate https://github.com/weaviate/weaviate-chaos-engineering/tree/main/apps/ann-benchmarks style format
+type ResultsJSONBenchmark struct {
+	Api              string  `json:"api"`
+	Ef               int     `json:"ef"`
+	EfConstruction   int     `json:"efConstruction"`
+	MaxConnections   int     `json:"maxConnections"`
+	Mean             float64 `json:"meanLatency"`
+	P99Latency       float64 `json:"p99Latency"`
+	QueriesPerSecond float64 `json:"qps"`
+	Shards           int     `json:"shards"`
+	Parallelization  int     `json:"parallelization"`
+	Limit            int     `json:"limit"`
+	ImportTime       float64 `json:"importTime"`
+	RunID            string  `json:"run_id"`
```

Comment:
Do we have both camelCase and snake case or is it a typo?

Reply:
@asdine good point. I kept the inconsistent casing of `run_id` / `maxConnections` etc to match the json format of the original python chaos benchmarks.  I'd like to keep the same keys so we can plot both results.